Library/Homebrew/cmd/audit.rb

```diff
@@ -238,13 +238,40 @@ def audit_urls
       problem "Savannah homepages should be https:// links (URL is #{homepage})."
     end
 
+    if homepage =~ %r[^http://((?:trac|tools|www)\.)?ietf\.org]
+      problem "ietf homepages should be https:// links (URL is #{homepage})."
+    end
+
+    if homepage =~ %r[^http://((?:www)\.)?gnupg.org/]
+      problem "GnuPG homepages should be https:// links (URL is #{homepage})."
+    end
+
+    # Freedesktop is complicated to handle - It has SSL/TLS, but only on certain subdomains.
+    # To enable https Freedesktop change the url from http://project.freedesktop.org/wiki to
+    # https://wiki.freedesktop.org/project_name.
+    # "Software" is redirected to https://wiki.freedesktop.org/www/Software/project_name
+    if homepage =~ %r[^http://((?:www|nice|libopenraw|liboil|telepathy|xorg)\.)?freedesktop\.org/(?:wiki/)?]
```

Comment:
It needs to be contained. Not every subdomain is covered - It seems Freedesktop offer two options to individual projects, either to use a Freedesktop standard domain, which is then SSL/TLS covered, or to create your own project homepage which isn't covered by SSL/TLS. Theoretically we could be less strict than this, but I think we're less likely to run into problems if we're overly careful.